pkg/actions/lua/storage/gcloud/gcs.go

```diff
@@ -0,0 +1,131 @@
+package gcloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/Shopify/go-lua"
+	"github.com/treeverse/lakefs/pkg/actions/lua/path"
+	"github.com/treeverse/lakefs/pkg/actions/lua/util"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+)
+
+const (
+	// googleAuthCloudPlatform - Cloud Storage authentication https://cloud.google.com/storage/docs/authentication
+	googleAuthCloudPlatform = "https://www.googleapis.com/auth/cloud-platform"
+)
+
+var (
+	ErrInvalidGCSURI = errors.New("invalid Google Cloud Storage URI")
+)
+
+func Open(l *lua.State, ctx context.Context) {
+	open := func(l *lua.State) int {
+		lua.NewLibrary(l, []lua.RegistryFunction{
+			{Name: "gs_client", Function: newGSClient(ctx)},
+		})
+		return 1
+	}
+	lua.Require(l, "gcloud", open, false)
+	l.Pop(1)
+}
+
+func newGSClient(ctx context.Context) lua.Function {
+	return func(l *lua.State) int {
+		json := lua.CheckString(l, 1)
+		c := &GSClient{
+			JSON: json,
+			ctx:  ctx,
+		}
+		l.NewTable()
+		for name, goFn := range functions {
+			// -1: tbl
+			l.PushGoFunction(goFn(c))
+			// -1: fn, -2:tbl
+			l.SetField(-2, name)
+		}
+		return 1
+	}
+}
+
+type GSClient struct {
+	JSON string
+	ctx  context.Context
+}
+
+func (c *GSClient) client() (*storage.Client, error) {
+	var opts []option.ClientOption
+	cred, err := google.CredentialsFromJSON(c.ctx, []byte(c.JSON), googleAuthCloudPlatform)
+	if err != nil {
+		return nil, err
+	}
+	opts = append(opts, option.WithCredentials(cred))
+	return storage.NewClient(c.ctx, opts...)
```

Comment:
not using dynamic `opts` - just call `storage.NewClient(c.ctx, option.WithCredentials(cred))`

Reply:
True. Done